packages/DataTable/src/DataTable.tsx

```diff
@@ -72,29 +70,36 @@ export interface DataTableProps {
   /** Override the table Column configuration for some particular rows */
   renderRow?: ((row: Row<TableDataItemType>) => React.ReactNode) | null;
 
+  /** If the table should be resized when the viewport size changed */
+  shouldResizeWithViewport?: boolean;
+
   /** Experimental prop */
   extraCellProps?: Record<string, any>;
 
   [x: string]: any;
 }
 
-const DataTable: React.FC<DataTableProps> & DataTableComposition = ({
-  a11yText,
-  children = null,
-  columns,
-  data,
-  height,
-  width,
-  borderType = gridTypes.HORIZONTAL,
-  getRowHeight = null,
-  hasZebraStripes = false,
-  isHeaderSticky = true,
-  renderRow = null,
-  extraCellProps = {},
-  ...moreProps
-}: DataTableProps) => {
+function DataTable(
+  {
+    a11yText,
+    children = null,
+    columns,
+    data,
+    height = "100vh",
+    width = "100vw",
+    borderType = gridTypes.HORIZONTAL,
+    getRowHeight = null,
+    hasZebraStripes = false,
+    isHeaderSticky = true,
+    renderRow = null,
+    shouldResizeWithViewport = false,
```

Comment:
Maybe this should have a default of `true`? I guess it might decrease performance in some cases when it's not needed, but how often do you think it will not be needed compared to how often it will be? I _suspect_ it will be needed most of the time...?

Reply:
Yeah you're right, based on our design we always need it